### PR TITLE
Add missing tab-completion help for emoji

### LINKED ISSUE #34006: missing tab-completion help for emoji
As reported [on discourse](https://discourse.julialang.org/t/hover-over-latex-unicode-character-to-see-the-latex-sourcecode/31749/3?u=stevengj), the REPL help doesn't show tab completion info for emoji even when it exists, unlike for other Unicode completions:
```jl
help?> α
"α" can be typed by \alpha<tab>
search:

Couldn't find α
Perhaps you meant !, %, &, *, +, -, /, :, <, >, \, ^, |, ~, ÷, π, ℯ, ∈, ∉ or ∋
  No documentation found.

  Binding α does not exist.

help?> 👍
search:

Couldn't find 👍
Perhaps you meant !, %, &, *, +, -, /, :, <, >, \, ^, |, ~, ÷, π, ℯ, ∈, ∉ or ∋
  No documentation found.

  Binding 👍 does not exist.
```
even though `👍` is produced by tab completion of `\:+1:`

### PR DESCRIPTION
**Pull request by utkarsh2102**
Closes: #34006

### CODE REVIEW

**Comment by utkarsh2102** [comment]
Hi @stevengj,  
Could you review it?

**Comment by stevengj** [comment]
Could use a test.

**Comment by stevengj** [comment]
Note that to add a test, you can test the help string with, for example:
```jl
import REPL

@test startswith(let buf = IOBuffer()
        Core.eval(Main, REPL.helpmode(buf, "α"))
        String(take!(buf))
    end, "\"α\" can be typed by \\alpha<tab>\n")
```
and similarly for `🐨` .

File: stdlib/REPL/src/docview.jl [hunk 1]
```diff
@@ -240,7 +240,8 @@ repl_corrections(s) = repl_corrections(stdout, s)
 const symbols_latex = Dict{String,String}()
 function symbol_latex(s::String)
     if isempty(symbols_latex) && isassigned(Base.REPL_MODULE_REF)
-        for (k,v) in Base.REPL_MODULE_REF[].REPLCompletions.latex_symbols
```
Comment (KristofferC) [inline stdlib/REPL/src/docview.jl]: Couldn't the whole `Base.REPL_MODULE_REF[].` be removed?

Comment (stevengj) [reply stdlib/REPL/src/docview.jl]: Probably?  I guess that is an artifact of the old days when REPLCompletions was not part of the REPL module?

Comment (utkarsh2102) [reply stdlib/REPL/src/docview.jl:243]: Hm, right. Removed now.

**Comment by utkarsh2102** [comment]
Hi @stevengj @KristofferC,
Could you take a look at this now?
(very sorry for the delay :/)

File: stdlib/REPL/test/docview.jl [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+@test startswith(let buf = IOBuffer()
+        Core.eval(Main, REPL.helpmode(buf, "α"))
+        String(take!(buf))
+    end, "\"α\" can be typed by \\alpha<tab>\n")
+
+@test startswith(let buf = IOBuffer()
+        Core.eval(Main, REPL.helpmode(buf, "🐨"))
+        String(take!(buf))
+    end, "\"🐨\" can be typed by \\coala<tab>\n")
```
Comment (stevengj) [inline stdlib/REPL/test/docview.jl:13]: "koala", not "coala"?

Comment (stevengj) [reply stdlib/REPL/test/docview.jl:13]: And should also have colons:  `\:koala:<tab>`

Comment (utkarsh2102) [reply stdlib/REPL/test/docview.jl:13]: Oh my bad. Fixed! :smile:

File: stdlib/REPL/test/docview.jl [hunk 3]
```diff
@@ -5,9 +5,9 @@ import REPL
...
         Core.eval(Main, REPL.helpmode(buf, "α"))
         String(take!(buf))
-    end, "\"α\" can be typed by \\alpha<tab>\n")
+    end, "\"α\" can be typed by \:alpha<tab>\n")
 
 @test startswith(let buf = IOBuffer()
         Core.eval(Main, REPL.helpmode(buf, "🐨"))
         String(take!(buf))
-    end, "\"🐨\" can be typed by \\coala<tab>\n")
+    end, "\"🐨\" can be typed by \:koala<tab>\n")
```
Comment (stevengj) [inline stdlib/REPL/test/docview.jl:13]: You still need a double backslash, to [escape the backslash](https://xkcd.com/1638/).  And there needs to be a colon at the end `koala` as well as the beginning.

That is, it should be `\\:koala:`

File: stdlib/REPL/test/docview.jl [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+import REPL
+
+@test startswith(let buf = IOBuffer()
+        Core.eval(Main, REPL.helpmode(buf, "α"))
+        String(take!(buf))
+    end, "\"α\" can be typed by \\:alpha:<tab>\n")
```
Comment (stevengj) [inline stdlib/REPL/test/docview.jl:8]: No, there are only colons in the emoji completion (e.g. `:koala:`), not in the Greek letters.

In general, I would suggest trying out your tests locally to cut down on the back-and-forth.

**Comment by stevengj** [comment]
Rebased.

**Review by stevengj — APPROVED** [review]

**Comment by stevengj** [comment]
After I fixed the test script, it passes locally for me and this looks good to merge.

**Comment by stevengj** [comment]
I'm just going to go ahead and merge this since it's just documentation and pretty uncontroversial.

**Pull request merged**